src/redis_cmd.cc

```diff
@@ -3270,57 +3270,72 @@ class CommandPublish : public Commander {
   }
 };
 
+void SubscribeCommmandReply(std::string *output, std::string name, std::string sub_name, int num) {
+  output->append(Redis::MultiLen(3));
+  output->append(Redis::BulkString(name));
+  output->append(sub_name.empty() ? Redis::NilString() : Redis::BulkString(sub_name));
+  output->append(Redis::Integer(num));
+}
+
 class CommandSubscribe : public Commander {
  public:
-  CommandSubscribe() : Commander("subcribe", -2, false) {}
+  CommandSubscribe() : Commander("subscribe", -2, false) {}
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
     for (unsigned i = 1; i < args_.size(); i++) {
       conn->SubscribeChannel(args_[i]);
-      output->append(Redis::MultiLen(3));
-      output->append(Redis::BulkString("subscribe"));
-      output->append(Redis::BulkString(args_[i]));
-      output->append(Redis::Integer(conn->SubscriptionsCount()));
+      SubscribeCommmandReply(output, "subscribe", args_[i],
+        conn->SubscriptionsCount() + conn->PSubscriptionsCount());
     }
     return Status::OK();
   }
 };
 
 class CommandUnSubscribe : public Commander {
  public:
-  CommandUnSubscribe() : Commander("unsubcribe", -1, false) {}
+  CommandUnSubscribe() : Commander("unsubscribe", -1, false) {}
   Status Execute(Server *svr, Connection *conn, std::string *output) override {
-    if (args_.size() > 1) {
-      conn->UnSubscribeChannel(args_[1]);
-    } else {
+    if (args_.size() == 1) {
       conn->UnSubscribeAll();
+      SubscribeCommmandReply(output, "unsubscribe", "",
+        conn->SubscriptionsCount() + conn->PSubscriptionsCount());
```

Comment:
in unsubscribe all, a message for every unsubscribed channel will be sent to the client in stock redis (verified on redis v5.0.4)， is this response on purpose？